### ISSUE TITLE
Allow browsers to wrap long URL

### ISSUE THREAD

**Issue by manueljacob** [asciidoctor/asciidoctor#3544]
When having a long URL (like the following
```
https://example.com/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf/asdf
```
) in Asciidoc, the corresponding line in the output is wrapped by Firefox before a slash, but not by Chromium (but it would be wrapped after a question mark if the URL contained a query).

I think it would be best to allow the URL to wrap at any place (by using `overflow-wrap: break-word` CSS). Of course it’s not a particulary nice solution, but the problem can’t really be solved in a nice way. At least it should be made consistent, instead of being dependent of what the browser happens to recognize as word boundaries.

**Comment by mojavelinux** [comment]
Since Firefox applies break-word as it is, I think it's safe to add this setting to the whole document. That will allow it to inherit. It can then be overridden using the nobreak or nowrap roles.

We could start by applying this change to only the bare class, which is added to any bare URL. Though if we really want the browsers to behave the same, the setting needs to go on the document itself.

**Comment by mojavelinux** [comment]
I've decided to move forward with setting `word-wrap: anywhere` on the body since that seems to be what Firefox does by default. I've also loosened the selector for the nobreak (and nowrap) roles so they can be applied anywhere (such as to a URL) instead of just an inline code element.

**Comment by mojavelinux** [comment]
This change will also fix table cells that cause the same kind of overflow. It also retains the fix for #1814.
